app/handlers/GuaranteeBalanceResponseHandler.scala

```diff
@@ -146,11 +144,39 @@ class GuaranteeBalanceResponseHandler @Inject() (
     )
   }
 
+  private def auditSuccess()(implicit hc: HeaderCarrier, ec: ExecutionContext, request: DataRequest[_]): Unit =
+    auditService.audit(
+      UnsuccessfulBalanceAuditModel.build(
+        AUDIT_TYPE_GUARANTEE_BALANCE_RATE_LIMIT,
+        request.userAnswers.get(EoriNumberPage).getOrElse("-"),
+        request.userAnswers.get(GuaranteeReferenceNumberPage).getOrElse("-"),
+        request.userAnswers.get(AccessCodePage).getOrElse("-"),
+        request.internalId,
+        LocalDateTime.now,
+        TOO_MANY_REQUESTS,
+        ErrorMessage(AUDIT_ERROR_RATE_LIMIT_EXCEEDED, AUDIT_DEST_RATE_LIMITED)
+      )
+    )
+
+  private def auditRateLimit()(implicit hc: HeaderCarrier, ec: ExecutionContext, request: DataRequest[_]): Unit =
+    auditService.audit(
+      UnsuccessfulBalanceAuditModel.build(
+        AUDIT_TYPE_GUARANTEE_BALANCE_RATE_LIMIT,
+        request.userAnswers.get(EoriNumberPage).getOrElse("-"),
+        request.userAnswers.get(GuaranteeReferenceNumberPage).getOrElse("-"),
+        request.userAnswers.get(AccessCodePage).getOrElse("-"),
+        request.internalId,
+        LocalDateTime.now,
+        TOO_MANY_REQUESTS,
+        ErrorMessage(AUDIT_ERROR_RATE_LIMIT_EXCEEDED, AUDIT_DEST_RATE_LIMITED)
+      )
+    )
+
```

Comment:
Both 
private def auditService
 and 
private def auditRateLimit 
do exactly the same thing the code if identical